clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/common/cache/Keys.groovy

```diff
@@ -174,9 +182,15 @@ class Keys {
   }
 
   static String getClusterKey(AzureCloudProvider azureCloudProvider,
+                              String application,
                               String clusterName,
-                              String region,
                               String account) {
-    "${azureCloudProvider.id}:${Namespace.AZURE_CLUSTERS}:${clusterName}:${region}:${account}"
+    "${azureCloudProvider.id}:${Namespace.AZURE_CLUSTERS}:${application}:${clusterName}:${account}"
   }
+
+  static String getInstanceKey(AzureCloudProvider azureCloudProvider,
+                               String serverGroupName, String name, String region, String account) {
+    "${azureCloudProvider.id}:${Namespace.AZURE_INSTANCES.ns}:${serverGroupName}:${name}:${region}:${account}"
```

Comment:
Minor change: split the function params on multiple lines (just like the others); as long as the order is consistent with the one in the key, it makes it easer to spot typos.